src/libzippp.cpp

```diff
@@ -385,17 +431,12 @@ bool ZipArchive::setComment(const string& comment) const {
     return result==0;
 }
 
-bool ZipArchive::isEntryCompressionEnabled(const ZipEntry& entry) const {
-    return entry.compressionMethod==ZIP_CM_DEFLATE;
-}
-
-bool ZipArchive::setEntryCompressionEnabled(const ZipEntry& entry, bool value) const {
+bool ZipArchive::setEntryCompressionMethod(const ZipEntry& entry, CompressionMethod comp) const {
     if (!isOpen()) { return false; }
     if (entry.zipFile!=this) { return false; }
     if (mode==ReadOnly) { return false; }
-    
-    libzippp_uint16 compMode = value ? ZIP_CM_DEFLATE : ZIP_CM_STORE;
-    return zip_set_file_compression(zipHandle, entry.index, compMode, 0)==0;
+    const libzippp_uint16 comp_libzip = convertCompressionToLibzip(comp);
```

Comment:
Looks like there is a `entry.compressionMethod = comp` missing here.

Reply:
done